sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/trace/_utils.py

```diff
@@ -320,3 +339,65 @@ def _get_url_for_http_request(attributes: Attributes) -> Optional[str]:
                     http_target,
                 )
     return url
+
+def _get_DJB2_sample_score(trace_id_hex: str) -> float:
+    # This algorithm uses 32bit integers
+    hash_value = Int32(_SAMPLING_HASH)
+    for char in trace_id_hex:
+        hash_value = ((hash_value << 5) + hash_value) + ord(char)
+
+    if hash_value == _INTEGER_MIN:
+        hash_value = int(_INTEGER_MAX)
+    else:
+        hash_value = abs(hash_value)
+
+    # divide by _INTEGER_MAX for value between 0 and 1 for sampling score
+    return float(hash_value) / _INTEGER_MAX
+
+def _round_down_to_nearest(sampling_percentage: float) -> float:
+    if sampling_percentage == 0:
+        return 0
+    # Handle extremely small percentages that would cause overflow
+    if sampling_percentage <= _INTEGER_MIN:  # Extremely small threshold
+        return 0.0
+    item_count = 100.0 / sampling_percentage
+    # Handle case where item_count is infinity or too large for math.ceil
+    if not math.isfinite(item_count) or item_count >= _INTEGER_MAX:
+        return 0.0
+    return 100.0 / math.ceil(item_count)
+
+def parent_context_sampling(
+    parent_context: Optional[Context],
+    attributes: Attributes = None
+) -> Optional["SamplingResult"]:
+    parent_span = get_current_span(parent_context)
+    parent_span_context = parent_span.get_span_context()
+
+    if parent_context is not None:
```

Comment:
This check should be at the top, you don't need to run other code when parent_context = None